subprojects/performance/templates.gradle

```diff
@@ -272,6 +272,17 @@ task bigCppMulti(type: CppMultiProjectGeneratorTask) {
     subProjectTemplates = ['cpp-source', 'cpp-project']
 }
 
+tasks.withType(CppMultiProjectGeneratorTask) { task ->
+    if (!task.name.endsWith('WithMacroIncludes')) {
```

Comment:
It would be much nicer if we declared the projects to create as some kind of value objects, and iterated them and created the two different types of tasks based on the value objects. Something like this:

```groovy
class ProjectToGenerate {
  int projects
  int sourceFiles
  Map<String, Object> templateArgs
  List<String> subProjectTemplates
}

def projectsToGenerate = [
  bigCppMulti: new ProjectToGenerate(
    projects: 201,
    sourceFiles: 8,
    templateArgs: [
      functionCount: 3
    ],
    subProjectTemplates: ['cpp-source', 'cpp-project']
  )
  // ...
]

projectsToGenerate.each { name, params ->
  tasks.create(name, CppMultiProjectGeneratorTask) {
    projects = params.projects,
    // ...
  }
}
```

Reply:
I think we should resist making another way to define performance projects (we have two completely different ways right now).  We only need two more tasks and I think it's OK and probably a little clearer to just put these near the non-macro perf projects that already exist.